lib/core/src/Cardano/Wallet.hs

```diff
@@ -969,6 +985,119 @@ listTransactions ctx wid mStart mEnd order = db & \DBLayer{..} -> do
         -- the start of a slot and its end could be a valid candidate.
         txTime = slotStartTime sp
 
+createCert
+    :: forall ctx s t k.
+        ( HasTransactionLayer t k ctx
+        , HasLogger ctx
+        , HasDBLayer s k ctx
+        )
+    => ctx
+    -> WalletId
+    -> ExceptT ErrCreateCert IO CoinSelection
+createCert ctx wid = do
+    (wal, _, pending) <- withExceptT ErrCreateCertNoSuchWallet $
+        readWallet @ctx @s @k ctx wid
+    let bp = blockchainParameters wal
+    let utxo = availableUTxO @s pending wal
+    let sel = CoinSelection [] [] []
+    withExceptT ErrCreateCertFee $ do
+        debug tr "Coins selected for delegation certificate after fee adjustment"
+            =<< adjustForFee (feeOpts tl computeCertFee (bp ^. #getFeePolicy)) utxo sel
+  where
+    tl = ctx ^. transactionLayer @t @k
+    tr = ctx ^. logger
+
+signCert
+    :: forall ctx s t k.
+        ( HasTransactionLayer t k ctx
+        , HasDBLayer s k ctx
+        , Show s
+        , NFData s
+        , IsOwned s k
+        , GenChange s
+        , HardDerivation k
+        , AddressIndexDerivationType k ~ 'Soft
+        )
+    => ctx
+    -> WalletId
+    -> ArgGenChange s
+    -> Passphrase "encryption"
+    -> CoinSelection
+    -> PoolId
+    -> ExceptT ErrSignCert IO (Tx, TxMeta, UTCTime, [TxWitness])
+signCert ctx wid argGenChange pwd
+    (CoinSelection ins outs chgs) _poolId = db & \DBLayer{..} -> do
+    withRootKey @_ @s ctx wid pwd ErrSignCertWithRootKey $ \xprv -> do
+        mapExceptT atomically $ do
+            cp <- withExceptT ErrSignCertNoSuchWallet $ withNoSuchWallet wid $
+                readCheckpoint (PrimaryKey wid)
+            let (changeOuts, s') = flip runState (getState cp) $
+                    forM chgs $ \c -> do
+                        addr <- state (genChange argGenChange)
+                        return $ TxOut addr c
+            allShuffledOuts <- liftIO $ shuffle (outs ++ changeOuts)
```

Comment:
The shuffling is so-to-speak pointless in this scenario because all outputs are already ours. In the case of standard transaction, we shuffle to add some privacy and avoid having outputs that follow a very specific pattern:

- recipient
- change
- recipient
- change
etc ...

Here, there's no need :)

Reply:
indeed, corrected